### PR TITLE
bin.str.enc=utf8: Assume ASCII first 🇺🇸 + add some more tests

### PR DESCRIPTION
**Pull request by kazarmy**
**Your checklist for this pull request**
- [X] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [X] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [X] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

~~Sorry forgot about utf8 when testing `bin.str.enc`.~~ This pr fixes `bin.str.enc=utf8` so that all-ASCII strings are designated `ascii`.

(04f11f38b11a71a77ce830d316164a6d439025fa does mean that `string_scan_range()` treats `bin.str.enc=ascii` and `bin.str.enc=utf8` the same. One day `ascii`, `latin1` and `utf8` should be treated differently.)

**Test plan**

GitHubCI and AppVeyor are green.

**Closing issues**

...

### CODE REVIEW

File: test/db/cmd/cmd_i [hunk 1]
```diff
@@ -3536,13 +3536,17 @@ wz k\0i\0abcdefgh
...
+.(izzenc utf8)
+?e
 .(izzenc utf16le)
 EOF
 EXPECT=<<EOF
 0   0x00000000 0x00000000 6   14           utf16le ki扡摣晥桧 blocks=Basic Latin,CJK Unified Ideographs
 
 0   0x00000004 0x00000004 8   9            ascii abcdefgh
 
+0   0x00000004 0x00000004 8   9            utf8 abcdefgh
```
Comment (radare) [inline test/db/cmd/cmd_i:3548]: why  is 'abcdefgh' considered utf8?

Comment (kazarmy) [reply test/db/cmd/cmd_i:3548]: because `e bin.str.enc=utf8` forces `string_scan_range()` to see only utf8 strings

Comment (kazarmy) [reply test/db/cmd/cmd_i:3548]: and it is a valid utf8 string

**Comment by radare** [comment]
i would consider this string ascii because it contains no non-ascii chars

**Comment by radare** [comment]
can we make strings containing only ascii chars to be considered ascii?

**Comment by kazarmy** [comment]
yes probably

**Comment by kazarmy** [comment]
Note that 04f11f38b11a71a77ce830d316164a6d439025fa does mean that `string_scan_range()` treats `bin.str.enc=ascii` and `bin.str.enc=utf8` the same.

One day `ascii`, `latin1` and `utf8` should be treated differently.

**Comment by kazarmy** [comment]
pr title will probably look better on Linux compared to Windows

**Comment by radare** [comment]
lol with the emoji

**Comment by kazarmy** [comment]
merge it in now?

**Pull request merged**